AppGWBEHealthVMSS/checkConcurrentCons.cs

```diff
@@ -150,45 +150,96 @@ public static void doCheck(ILogger log)
                 {
-                    log.LogInformation("Logging Custom Metrics");
-                    metricJSONGenerator.populateCustomMetric("RPS", rps.ToString(), appGw.Name, scaleSet.RegionName, scaleSet.Id, log, clientID, clientSecret, tenantID, logCustomMetricsVerboseLogging);
-                    metricJSONGenerator.populateCustomMetric("IdealNodeCount", idealNumberOfNodes.ToString(), appGw.Name, scaleSet.RegionName, scaleSet.Id, log, clientID, clientSecret, tenantID, logCustomMetricsVerboseLogging);
-                    metricJSONGenerator.populateCustomMetric("MaxConcurrentConnectionsPerNode", maxConcurrentConnectionsPerNode.ToString(), appGw.Name, scaleSet.RegionName, scaleSet.Id, log, clientID, clientSecret, tenantID, logCustomMetricsVerboseLogging);
+                    if (logCustomMetrics)
+                    {
+                        if (allowedMetricRegions.Split(",").Contains(scaleSet.RegionName))
+                        {
+                            log.LogInformation("Logging Custom Metrics");
+                            metricJSONGenerator.populateCustomMetric("RPS", rps.ToString(), appGw.Name, scaleSet.RegionName, scaleSet.Id, log, clientID, clientSecret, tenantID, logCustomMetricsVerboseLogging);
+                            metricJSONGenerator.populateCustomMetric("IdealNodeCount", idealNumberOfNodes.ToString(), appGw.Name, scaleSet.RegionName, scaleSet.Id, log, clientID, clientSecret, tenantID, logCustomMetricsVerboseLogging);
+                            metricJSONGenerator.populateCustomMetric("MaxConcurrentConnectionsPerNode", maxConcurrentConnectionsPerNode.ToString(), appGw.Name, scaleSet.RegionName, scaleSet.Id, log, clientID, clientSecret, tenantID, logCustomMetricsVerboseLogging);
+                        }
+                        else
+                        {
+                            log.LogInformation("Skipping custom metric population, as we're not in a region that supports it (defined in appSetting)");
+                        }
+                    }
+                    else
+                    {
+                        log.LogInformation($"Skipping logging Custom Metrics per appSetting");
+                    }
                 }
-                else
+                catch (Exception customLoggingError)
                 {
-                    log.LogInformation($"Skipping logging Custom Metrics per appSetting");
+                    log.LogInformation($"Error with custom metric population - continuing anyway as this was Plan B {customLoggingError}");
                 }
-                // If we will scale down, hold off unless we get a consistent message to do that
-                int idealNodes = (int)Math.Ceiling(idealNumberOfNodes);
-                if (idealNumberOfNodes < scaleSet.Capacity)
+
+                //Add check if autoscale configured - if so, don't custom scale
+                bool scaleSetAutoScaleRuleIsEnabled = false;
+                try
                 {
```

Comment:
this whole thing seems a bit expensive to check every time around the loop.  will it ever change from run to run?  Seems like we should just remember whether or not we have auto scale enabled once and remember it.  Or maybe check every N times rather than every time.  Pushing into a method would also be nice to keep things from getting even more monolithic in this function.